netsim/sniffer/process.py

```diff
@@ -0,0 +1,402 @@
+        box_name = draw.Text(node['type'], 16, x-self.offset_x, y+10, center=True, fill='#fff', font_family='Helvetica')
+        res.append(box)
+        res.append(box_name)
+        if self.show_node_labels:
+            label = draw.Text(node['id'], 14, x-self.offset_x, y+40, center=True, fill='#666', font_family='Helvetica')
+            res.append(label)
+            ipls = ''
+            if node['id'] != node['ip']:
+                ipls = node['ip']
+            ip_label = draw.Text(ipls, 14, x-self.offset_x, y+60, center=True, fill='#666', font_family='Helvetica')
+            res.append(ip_label)
+        self.animate_node_to(res, x, y, 0)
+        self.animate_node_to(res, x, y, 0.5)
+        return res
+
+    def draw_pkt(self, x, y, pkt, t, id):
+        res = []
+        stroke = None
+        if pkt['ipv6']:
+            stroke = '#d2a'
+        else:
+            stroke = '#84a59d'
+        circle = draw.Circle(x-self.offset_x, y, 0, fill=self.pkt_color_map[pkt['type']], stroke=stroke, stroke_width=2, class_='pkt', data_label=pkt['type'], data_id=id)
+        res.append(circle)
+        self.animate_pkt_to(res, x, y, 0, t)
+        return res
+
+    def animate_node_to(self, node, x, y, t):
+        node[0].add_key_frame(t, x=x-self.node_size/2-self.offset_x, y=y-self.node_size/2)
+        node[1].add_key_frame(t, x=x-self.offset_x, y=y+10)
+        i = 2
+        if self.show_node_labels:
+            node[i].add_key_frame(t, x=x-self.offset_x, y=y+40)
+            node[i+1].add_key_frame(t, x=x-self.offset_x, y=y+60)
+            i+=2
+
+    def animate_pkt_to(self, pkt, x, y, r, t):
+        pkt[0].add_key_frame(t, cx=x-self.offset_x, cy=y, r=r)
+
+    def show_pkt(self, pkt, t):
+        pkt[0].add_key_frame(t+self.batch_interval/10, r=self.pkt_size)
+
+    def hide_pkt(self, pkt, t):
+        pkt[0].add_key_frame(t, r=0)
+
+    def send_pkt(self, n_from, n_to, pkt, t, i, pkt_id):
+        p_from = n_from['pos']
+        p_to = n_to['pos']
+        p = self.draw_pkt(p_from[0], p_from[1], pkt, t, pkt_id)
+        self.show_pkt(p, t)
+        self.animate_pkt_to(p, p_to[0], p_to[1], self.pkt_size, t+i)
+        self.hide_pkt(p, t+i)
+        return p
+
+    def calculate_node_position(self, r, node_cnt, i):
+        x = r*math.cos(i*2*math.pi/node_cnt)-self.offset_x
+        y = r*math.sin(i*2*math.pi/node_cnt)
+        return (x, y)
+    
+def run_viz(args):
+    t_stdout = sys.stdout
+    class PseudoNonTTY(object):
+        def __init__(self, underlying):
+            self.__underlying = underlying
+        def __getattr__(self, name):
+            return getattr(self.__underlying, name)
+        def isatty(self):
+            return False
+    
+    sys.stdout = PseudoNonTTY(sys.stdout) # disable color output on packet data for JS
+    
+    viz = NetsimViz()
+    viz.run(args)
+    viz.export(args['output'])
+
+    sys.stdout = t_stdout
```

Comment:
This should all be in a `try... except` block so that you restore `sys.stdout` whatever happens.